PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -196,35 +199,29 @@ function transformReports(inputReportFiles: string[], index: number)
         saxStream.on("end", function() {
             switch (reportType) {
                 case ReportType.XML_STATIC:
-                    transformToSarif(report);
+                    transformToSarif(report, ruleDocUrlPromises.length > 0);
                     break;
                 case ReportType.XML_TESTS:
                     transformToXUnit(report);
                     break;
                 case ReportType.XML_STATIC_AND_TESTS:
-                    transformToSarif(report);
+                    transformToSarif(report, ruleDocUrlPromises.length > 0);
                     transformToXUnit(report);
                     break;
                 case ReportType.XML_SOATEST:
                     transformToSOATestXUnit(report);
                     break;
                 case ReportType.XML_STATIC_AND_SOATEST:
                     transformToSOATestXUnit(report);
-                    transformToSarif(report);
+                    transformToSarif(report, ruleDocUrlPromises.length > 0);
```

Comment:
Instead of checking "ruleDocUrlPromises" inside "transform" method, I passed the boolean value in here because I only want it to go through the logic when transforming to a SARIF report.